### PR TITLE
added uidtool

### PR DESCRIPTION
**Pull request by Akshat2Jain**
this pr is related to https://github.com/plone/Products.CMFPlone/pull/3836

### CODE REVIEW

**Review by wesleybl — CHANGES_REQUESTED** [review]
I think the commit message should be something like:

"Prevents handleUidAnnotationEvent failure if portal_uidhandler is not available"

Please add an entry to Changes

File: CHANGES.rst [hunk 1]
```diff
@@ -4,6 +4,7 @@ Products.CMFUid Changelog
 4.1 (unreleased)
 ----------------
 
+- modified the code to check if both 'anno_tool' and 'uidtool' exist before using them. This change ensures that the code won't run unless both tools are available, preventing the 'NoneType' error.
```
Comment (wesleybl) [inline CHANGES.rst:7]:
```suggestion
- Modified the code of ``handleUidAnnotationEvent`` to check if both 'anno_tool' and 'uidtool' exist before using them. This change ensures that the code won't run unless both tools are available, preventing the 'NoneType' error.
```

**Review by wesleybl — CHANGES_REQUESTED** [review]

**Review by wesleybl — APPROVED** [review]

**Comment by wesleybl** [comment]
@icemac @dataflake can you create a new release please?

**Comment by dataflake** [comment]
Version 4.1 is now released.

**Comment by wesleybl** [comment]
> Version 4.1 is now released.

Thanks!

**Pull request merged**